lib/new_relic/agent/instrumentation/grpc/server/instrumentation.rb

```diff
@@ -12,29 +12,32 @@ module GRPC
         module Server
           include NewRelic::Agent::Instrumentation::GRPC::Helper
 
+          DT_KEYS = [NewRelic::NEWRELIC_KEY, NewRelic::TRACEPARENT_KEY, NewRelic::TRACESTATE_KEY].freeze
           INSTANCE_VAR_HOST = :@host_nr
           INSTANCE_VAR_PORT = :@port_nr
           INSTANCE_VAR_METHOD = :@method_nr
 
-          def handle_with_tracing(active_call, mth, _inter_ctx)
+          def handle_with_tracing(streamer_type, active_call, mth, _inter_ctx)
             return yield unless trace_with_newrelic?
-            metadata = metadata_for_call(active_call)
 
-            finishable = NewRelic::Agent::Tracer.start_transaction_or_segment(
-              name: mth.original_name,
-              category: :web,
-              options: server_options(metadata)
-            )
+            metadata = metadata_for_call(active_call)
+            options = trace_options(metadata, streamer_type)
+            txn = NewRelic::Agent::Transaction.start_new_transaction(NewRelic::Agent::Tracer.state,
+              options[:category],
+              options)
             process_distributed_tracing_headers(metadata)
 
             begin
               yield
             rescue => e
```

Comment:
The exception we're rescuing here looks pretty rare and only one of two possible exceptions. It doesn't seem likely to be related to the content of the request. We'll capture and report those, but they'll never give us the status code.